### ISSUE TITLE
Why is there a margin-bottom on <Row /> if gutter is used as an array

### ISSUE THREAD

**Issue by chillyistkult** [ant-design/ant-design#28976]
- [x] I have searched the [issues](https://github.com/ant-design/ant-design/issues) of this repository and believe that this is not a duplicate.

### Reproduction link
[https://ant.design/components/grid/](https://ant.design/components/grid/)

<img width="890" alt="Bildschirmfoto 2021-01-21 um 16 27 21" src="https://user-images.githubusercontent.com/1689047/105373304-d64f1c00-5c06-11eb-9ff0-95f29069bb84.png">

### Steps to reproduce
Apply a [horizontal, vertical] gutter to `<Row />`

### What is expected?
Horizontal and vertical padding is applied to nested `<Col />'s` only.

### What is actually happening?
Horizontal and vertical padding is applied to nested `<Col />'s` AND in addition there is a `margin-bottom` set on the `<Row />`

| Environment | Info |
|---|---|
| antd | 4.10.3 |
| React | 16.13.1 |
| System | OS X 11.1 |
| Browser | Chrome 87.0.4280.141 (Offizieller Build) (x86_64) |

---
This behavior does not make sense to me. It's not a given that a developer wants to have vertical spacing between `<Row />` and anything else and in my opinion this should be removed or at least overridable. If the developer wants to have vertical spacing he could use `<Space />` or use custom styling.

**Comment by zombieJ** [comment]
ref https://github.com/ant-design/ant-design/pull/16635/files#diff-24039ff6148d169a887fe30e0328a9c8336c49bf909b8a355c128510dac4c7eaR105

**Comment by zombieJ** [comment]
Not sure why I changed this. Need more dig in.

**Comment by zombieJ** [comment]
Tested. When multiple row exist, it will leave a empty space since we use `margin` to keep the space.

![截屏2021-01-26 下午4 47 11](https://user-images.githubusercontent.com/5378891/105821747-2575c200-5ff6-11eb-9519-e1fc8c4868bd.png)

But it may keep the place when other element exist:
![截屏2021-01-26 下午4 50 45](https://user-images.githubusercontent.com/5378891/105822129-a339cd80-5ff6-11eb-8a99-ccbb3c878fbe.png)

Second one seems more reasonable for you.

**Comment by chillyistkult** [comment]
@zombieJ Maybe I don't understand your answer correctly, but the margin is also added if there is only one Row with a horizontal and vertical gutter.

**Why is this a problem for me and probably others?**

I use a Row / Col setup in `PageHeader` extra to render some components (Buttons and Inputs). This works good, because I can easily adapt the Col breakpoints to render nicely for each screen size and control the spacing with the gutters. But because margin bottom is added to Row the components are not aligned anymore with the title, so I have to manually unset the margin.
